web/src/main/java/org/phoenixctms/ctsms/web/util/DateUtil.java

```diff
@@ -131,16 +131,22 @@ public static String getBookingDurationString(long duration) {
 	}
 
 	public static DateFormat getDateFormat() {
+		return getDateFormat(Settings.getBoolean(SettingCodes.DATE_USER_TIME_ZONE, Bundle.SETTINGS, DefaultSettings.DATE_USER_TIME_ZONE));
+	}
+
+	public static DateFormat getDateFormat(boolean isUserTimeZone) {
 		Locale locale = WebUtil.getLocale();
 		TimeZone timeZone = WebUtil.getTimeZone();
-		DateFormat dateFormat = null;
-		String dateformatDatePattern = Settings.getString(SettingCodes.DATEFORMAT_DATE_PATTERN, Bundle.SETTINGS, DefaultSettings.DATEFORMAT_DATE_PATTERN);
-		if (dateformatDatePattern != null && dateformatDatePattern.length() > 0) {
-			dateFormat = new SimpleDateFormat(dateformatDatePattern, locale);
-		} else {
-			dateFormat = DateFormat.getDateInstance(Settings.getInt(SettingCodes.DATEFORMAT_DATE_STYLE, Bundle.SETTINGS, DefaultSettings.DATEFORMAT_DATE_STYLE), locale);
+		DateFormat dateFormat = new SimpleDateFormat(CommonUtil.getInputDatePattern(WebUtil.getDateFormat()), locale);
```

Comment:
Further down the line this dateFormat will be used in InputFieldValueStringAdapterBase. SimpleDateFormat is not threadsafe but I couldn't see any handling of this. Will this SimpleDateFormat (or others) will be used in a multithreaded context?

Reply:
- there is no threadsafe DateFormat
- yes in genereal one can presume the class will be accesses by multiple threads, although there are only few situations when instances will be singletons that will be shared by threads (see synchronized methods), so in general such classes are desinged thread safe.
-the pattern used here to circumvent using not-threadsafe Objects like DateFormat is to create a local instance if it